crates/ruff_python_parser/src/lexer.rs

```diff
@@ -1373,15 +1373,33 @@ impl<'src> Lexer<'src> {
         }
 
         let mut current_position = self.current_range().start();
-        let reverse_chars = self.source[..current_position.to_usize()].chars().rev();
+        let mut reverse_chars = self.source[..current_position.to_usize()]
+            .chars()
+            .rev()
+            .peekable();
         let mut newline_position = None;
 
-        for ch in reverse_chars {
+        while let Some(ch) = reverse_chars.next() {
             if is_python_whitespace(ch) {
                 current_position -= ch.text_len();
             } else if matches!(ch, '\n' | '\r') {
                 current_position -= ch.text_len();
-                newline_position = Some(current_position);
+                // Count the number of backslashes before the newline character.
+                let mut backslash_count = 0;
+                while reverse_chars.next_if_eq(&'\\').is_some() {
+                    backslash_count += 1;
+                }
```

Comment:
Last comment :) Do we need to restrict the escape handling to cases where we know we're inside a string? Or wouldn't that work in case of an unterminated string literal? 

I'm not sure if it matters because the parser is already in an error recovery state when encountering an escaped `\` outside of a string, but it might be worth to add a test for it

```python
test + a \\\
more
```

Reply:
I'm not sure I follow here. Do you mean to ask whether this logic needs to be restricted to only recovering within a string or not? I don't think so that is necessary, I'll add a test case for line continuation character encountered while re-lexing outside of a string.